packages/react-vapor/src/components/validation/hoc/WithInitialValuesMultiSelectHOC.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IMultiSelectOwnProps} from '../../select/MultiSelectConnected';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationTypes} from '../ValidationTypes';
+import {InferableComponentEnhancer} from './connectHOC';
+
+const mapDispatchToProps = (dispatch: IDispatch, {id}: IMultiSelectOwnProps) => ({
+    setWarning: (warning: string) =>
+        dispatch(ValidationActions.setWarning(id, warning, ValidationTypes.wrongInitialValue)),
+    clearWarning: () => dispatch(ValidationActions.clearWarning(id, ValidationTypes.wrongInitialValue)),
+});
+
+export type IMultiSelectWithInitialValuesOwnProps = {
+    initialValues: string[];
+    invalidInitialValuesMessage?: (invalidValues: string[]) => string;
+    resetWarningOnUnmount?: boolean;
```

Comment:
Should the name of this props (`resetWarningOnUnmount`) reflect the fact that it will reset only the `invalidInitialValuesMessage`? In the case where we would create a component with multiple validations, the name of this prop would lead towards confusion as to what warning it will actually reset.